registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java

```diff
@@ -1,609 +1,726 @@
+					registrationStatusDto.getRegistrationId(), DemoDedupeConstants.DEMO_SUCCESS);
+		}
+		return duplicateDtos;
+	}
+
+	/**
+	 * Gets the latest transaction id.
+	 *
+	 * @param registrationId
+	 *            the registration id
+	 * @return the latest transaction id
+	 */
+	private String getLatestTransactionId(String registrationId) {
+		RegistrationStatusEntity entity = registrationStatusDao.findById(registrationId);
+		return entity != null ? entity.getLatestRegistrationTransactionId() : null;
+	}
+
+	/**
+	 * Process demo dedupe requesthandler.
+	 *
+	 * @param registrationStatusDto the registration status dto
+	 * @param object                the object
+	 * @param description
+	 * @return true, if successful
+	 * @throws ApisResourceAccessException           the apis resource access
+	 *                                               exception
+	 * @throws IOException                           Signals that an I/O exception
+	 *                                               has occurred.
+	 * @throws                                       io.mosip.kernel.core.exception.IOException
+	 * @throws PacketDecryptionFailureException
+	 * @throws RegistrationProcessorCheckedException
+	 */
+	private boolean processDemoDedupeRequesthandler(InternalRegistrationStatusDto registrationStatusDto,
+			MessageDTO object, LogDescription description) throws ApisResourceAccessException, IOException,
+			PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException,
+			RegistrationProcessorCheckedException, JsonProcessingException, PacketManagerException {
+		boolean isTransactionSuccessful = false;
+		List<String> responsIds = new ArrayList<>();
+
+		String latestTransactionId = getLatestTransactionId(registrationStatusDto.getRegistrationId());
+
+		List<AbisResponseDto> abisResponseDto = packetInfoManager.getAbisResponseRecords(latestTransactionId,
+				DemoDedupeConstants.IDENTIFY);
+
+		for (AbisResponseDto responseDto : abisResponseDto) {
+			if (responseDto.getStatusCode().equalsIgnoreCase(AbisStatusCode.SUCCESS.toString())) {
+				responsIds.add(responseDto.getId());
+			} else {
+				isTransactionSuccessful = true;
+				int retryCount = registrationStatusDto.getRetryCount() != null
+						? registrationStatusDto.getRetryCount() + 1
+						: 1;
+				description.setMessage(
+						DemoDedupeConstants.SENDING_TO_REPROCESS + " -- " + registrationStatusDto.getRegistrationId());
+				registrationStatusDto.setRetryCount(retryCount);
+
+				registrationStatusDto.setLatestTransactionStatusCode(registrationExceptionMapperUtil
+						.getStatusCode(RegistrationExceptionTypeCode.DEMO_DEDUPE_ABIS_RESPONSE_ERROR));
+				registrationStatusDto.setStatusComment(StatusUtil.DEMO_DEDUPE_FAILED_IN_ABIS.getMessage());
+				registrationStatusDto.setSubStatusCode(StatusUtil.DEMO_DEDUPE_FAILED_IN_ABIS.getCode());
+				registrationStatusDto.setStatusCode(RegistrationStatusCode.PROCESSING.toString());

```

Comment:
this change is correct, since the setLatestTransactionStatusCode will be reprocess, but lets check this specifically from Mono once and mention about this bug fix in the jira ticket also explictly.

Reply:
Confirmed with Mono,its fine changes